# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: gfx/wr/webrender/src/prim_store/gradient/radial.rs
# Commit: 4b7de1b38a00
# Full Hash: 4b7de1b38a008493b6c6fbb599049358f707e08f
# Author: Nicolas Silva <nsilva@mozilla.com>
# Date: 2021-04-27 15:52:11
# Description:
#   WIP: Bug 1707181 - Avoid zero-sized gradient tasks. r=gfx-reviewers,lsalzman
#   
#   Add missing check for linear gradients. Also, instead of just casting to int round out the task size.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D113289
# ==============================================================================

diff -r ee55297f011c -r 4b7de1b38a00 gfx/wr/webrender/src/prim_store/gradient/radial.rs
--- a/gfx/wr/webrender/src/prim_store/gradient/radial.rs	Tue Apr 27 07:41:16 2021 +0000
+++ b/gfx/wr/webrender/src/prim_store/gradient/radial.rs	Tue Apr 27 07:44:25 2021 +0000
@@ -166,7 +166,7 @@
             extend_mode: item.extend_mode,
             params,
             stretch_size,
-            task_size: task_size.to_i32(),
+            task_size: task_size.ceil().to_i32(),
             tile_spacing: item.tile_spacing.into(),
             brush_segments,
             stops_opacity,